### PR TITLE
Update stack-packet to compatibility with Crossplane v0.4.0

### LINKED ISSUE #8: Update stack-packet to be compatible with Crossplane v0.4.0
Crossplane [v0.4.0]() introduced a number of breaking changes including:
- Removal of portable resource classes [1]
- Moving all managed resources and resource classes to cluster-scoped [1]
- Introduction of claim scheduling controller and claim defaulting controllers [1]
- Introduction of `forProvider` / `atProvider` paradigm (not breaking) [2]
- Cross-resource references (not breaking, not applicable until reliant resources exist in this stack) [3]

The design of these components is reflected in the following docs:
[1] https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-simple-class-selection.md
[2] https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-managed-resource-api-design.md
[3] https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-cross-resource-referencing.md

`stack-packet` should be updated to be compatible with these changes.

### PR DESCRIPTION
**Pull request by hasheddan**
This updates `stack-packet` to be compatible with Crossplane v0.4.0. Notably, it does **NOT** update the apis to follow standards established for [optional fields](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#pointer-types-and-markers) and [high-fidelity / late initialization](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#high-fidelity). `TODO` comments have been added to denote where further improvements are needed.

Fixes #8

### CODE REVIEW

**Comment by hasheddan** [comment]
I can't assign as reviewers, but would love to get @negz and @muvaf eyes on this.

**Review by jasmingacic — APPROVED** [review]
Looking good

**Comment by jasmingacic** [comment]
Let's wait on @negz and @muvaf before we merge this.

File: pkg/controller/server/device/managed.go
Comment (negz) [inline pkg/controller/server/device/managed.go:177]: It seems like we're recording this here because devices have a non-deterministic ID we need to use to refer to them. The new and improved pattern here is to:

1. Disable the default external naming behaviour by providing your own intializer chain per discussion in 
https://github.com/crossplaneio/crossplane-runtime/pull/45
1. Record this ID as the managed resource's external name annotation via `meta.SetExternalName()`
1. Use `meta.GetExternalName()` rather than `d.Status.AtProvider.ID` when making calls to `e.client`.

**Review by negz — APPROVED** [review]
Looks good - one comment about the external naming strategy.

File: pkg/controller/server/device/managed.go
Comment (hasheddan) [reply pkg/controller/server/device/managed.go:177]: Ah yes, thank you for pointing this out!

File: pkg/controller/server/device/managed.go
Comment (hasheddan) [inline pkg/controller/server/device/managed.go:127]: Continuing to set the `ID` in status in addition to external name annotation as it is part of the device observation struct.

**Review by hasheddan — COMMENTED** [review]
@negz can you take a final pass at this?

**Comment by negz** [comment]
@hasheddan This looks good to me!

**Comment by jasmingacic** [comment]
@hasheddan Since we are on the same page. Should I merge this?

**Pull request merged**
